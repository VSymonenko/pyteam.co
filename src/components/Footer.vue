<template lang="pug">
  v-footer(app :style="{ padding: '10px'}")
    span &copy; 2018
    v-layout(justify-end)
      v-btn-toggle(v-model="text" mandatory class="v-footer")
        v-btn(flat v-for="(lang, index) in languages" :value="lang" :key="lang") {{ lang }}
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import { NoCache } from '@/decorators';

@Component({
  name: 'Footer',
  props: {
    language: {
      type: String,
    },
    languages: {
      type: Array,
    },
  },
})

export default class Footer extends Vue {
  @NoCache
  get text(): string {
    return this.$props.language;
  }

  set text(value: string) {
    this.$emit('set-language', value);
  }
}
</script>

<style lang="stylus">
.v-btn-toggle--selected
  box-shadow none
</style>
