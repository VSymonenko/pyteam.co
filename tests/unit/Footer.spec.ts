import Footer from '@/components/Footer.vue';
import { shallowMount } from '@vue/test-utils';
import './common';

test('renders correctly', () => {
  const wrapper = shallowMount(Footer);

  expect(wrapper.isVueInstance()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});

describe('Footer', () => {
  it('renders AppBar.language when passed', () => {
    const language = 'new message';
    const wrapper = shallowMount(Footer, {
      propsData: { language },
    });
    expect(wrapper.props().language).toContain(language);
  });

  it('renders a for each button in Footer', () => {
    const languages = ['1', '2'];
    const wrapper = shallowMount(Footer, {
      propsData: { languages },
    });
    expect(wrapper.props().languages).toMatchObject(languages);
  });
});
