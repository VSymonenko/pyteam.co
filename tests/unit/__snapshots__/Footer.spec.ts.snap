// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<vfooter-stub height="32" app="app" style="padding: 10px;"><span>Â© 2018</span>
  <vlayout-stub tag="div" justify-end="justify-end">
    <vbtntoggle-stub mandatory="true" class="v-footer"></vbtntoggle-stub>
  </vlayout-stub>
</vfooter-stub>
`;
